### PR TITLE
added delete_post_{->post_type} and deleted_post_{- >post_type} hook

### PR DESCRIPTION
**Pull request by benniledl**
added delete_post_{->post_type} and deleted_post_{- >post_type} hook

Trac ticket: https://core.trac.wordpress.org/ticket/60433

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by gerardreches** [comment]
@at-benni That should do 😄

File: src/wp-includes/post.php [hunk 1]
```diff
@@ -3555,6 +3555,16 @@ function wp_delete_post( $postid = 0, $force_delete = false ) {
 	 */
 	do_action( 'delete_post', $postid, $post );
 
+	/**
+	 * Fires immediately before a post is deleted from the database.
+	 *
+	 * @since 6.5.0
```
Comment (swissspidy) [inline src/wp-includes/post.php]:
```suggestion
	 * @since 6.6.0
```

File: src/wp-includes/post.php [hunk 2]
```diff
@@ -3555,6 +3555,16 @@ function wp_delete_post( $postid = 0, $force_delete = false ) {
 	 */
 	do_action( 'delete_post', $postid, $post );
 
+	/**
+	 * Fires immediately before a post is deleted from the database.
+	 *
```
Comment (swissspidy) [inline src/wp-includes/post.php:3560]:
```suggestion
	 *
	 * The dynamic portion of the hook name, `$post->post_type`, refers to
	 * the post type slug.
	 *
```

File: src/wp-includes/post.php [hunk 3]
```diff
@@ -3571,6 +3581,16 @@ function wp_delete_post( $postid = 0, $force_delete = false ) {
 	 */
 	do_action( 'deleted_post', $postid, $post );
 
+	/**
+	 * Fires immediately after a post is deleted from the database.
+	 *
+	 * @since 6.5.0
```
Comment (swissspidy) [inline src/wp-includes/post.php]:
```suggestion
	 * @since 6.6.0
```

File: src/wp-includes/post.php [hunk 4]
```diff
@@ -3571,6 +3581,16 @@ function wp_delete_post( $postid = 0, $force_delete = false ) {
 	 */
 	do_action( 'deleted_post', $postid, $post );
 
+	/**
+	 * Fires immediately after a post is deleted from the database.
+	 *
```
Comment (swissspidy) [inline src/wp-includes/post.php:3589]:
```suggestion
	 *
	 * The dynamic portion of the hook name, `$post->post_type`, refers to
	 * the post type slug.
	 *
```

**Review by audrasjb — APPROVED** [review]

**Comment by swissspidy** [comment]
Committed in https://core.trac.wordpress.org/changeset/57853

**Pull request closed without merging**